src/core/Gsim.cc

```diff
@@ -480,8 +480,8 @@ void Gsim::MakeEvent(const G4Event* g4ev, DS::Root* ds) {
 
   // Get the PMT type for IDPMTs. Then in the loop,
   // increment numPE only when the PE is in an IDPMT.
-  // Map from PMT ID numbers to objects for use later in noise calculation
-  std::map<int, DS::MCPMT*> mcpmtObjects;
+  //Map ID-INDEX for later noise calculation
+  std::map<int, int> mcpmtObjects;
```

Comment:
Ah, I just saw the comment. If there's a situation where the `rat_mcpmt` doesn't exist here, we should understand what causes that -- probably aborting is still the right thing to do!